### PR TITLE
feat(dev): CTC-403 inc 1 — a host resolves Linear team and state ids from the replica, not from hand-written caches

### PR DESCRIPTION
**Pull request by ryanrozich**
Part of CTC-403 (increment 1 — **additive**, no reader is cut over yet).

Identity is stored in **four** places, none authoritative. A Linear-to-Linear import
preserves team keys but **mints new internal UUIDs** — and Catalyst keys on the UUID while
caching on the key. So after a workspace cutover every cache still **matched** and every
value beneath it was dead. That is the worst combination: it defeats staleness detection
by construction.

The cloud registry is already authoritative for `(account → workspace → team)` and the
replica already carries every workflow state on every issue, so this is mostly
**subtraction** — read the copy that is already replicated.

## What's here

**`lib/linear-identity.sh`** — team id, state id, the full name→id map, and label id, read
from the replica behind the **same** `replica_fresh` gate the ticket-read helper uses
(writer-lock recency AND a non-empty `sync_meta` cursor). Reused rather than re-derived:
two gates that must agree, maintained separately, is exactly the drift this removes.

**`resolve-linear-ids.sh` asks the replica first.** Its API path needs `linear.apiToken`
in the per-project secrets file and **hard-exits without it**. On a host installed with a
Catalyst Cloud token and nothing else — the documented one-command install — that exit is
the whole reason team and state ids ended up hand-written across 21 config files. The API
stays as the fallback, and the source is **reported** either way.

## ⛔ The trap this deliberately does not walk into

Making identity a replica read couples it to replica freshness, and this fleet has that
failure on record. So every failure is **named** and there is **no stale fallback**:

| case | result | why not the obvious alternative |
| --- | --- | --- |
| empty **but fresh** replica | `state-not-found` | returning `{}` hands a caller a *confident nothing* — the CTL-1420 shape that froze admission fleet-wide with every signal green |
| replica absent vs stale | distinct reasons | they need different repairs; collapsing them turns the first diagnostic step into a guess |
| team with zero issues | `team-not-resolvable` | not "no such team" — indistinguishable from this source, and the stronger claim sends an operator to re-create a team that exists |
| duplicate label name | `label-ambiguous` | first-wins applies a label to the **wrong team's board** and looks like it worked |

## The `--json` stream contract

The fallback reason rides the **document** (`replicaFallbackReason`), not a bare stderr
line. `resolve-linear-ids.test.sh` runs the script with `> "$OUT" 2>&1` — stderr *merged*
into stdout, then piped to `jq` — so in `--json` mode a stderr note is not extra
diagnostics: it makes the whole document unparseable and every field unreadable. The first
cut shipped it as a stderr echo and broke three existing assertions.

Suppressing it under `--json` was the other option and is **worse**: the machine-readable
path is precisely where a silent degradation hides. Humans still get the stderr note.

```
$ resolve-linear-ids.sh --dry-run --json --force            # replica present
{"source":"replica","stateCount":12,"replicaFallbackReason":null}

$ CATALYST_REPLICA_DB=/nonexistent resolve-linear-ids.sh --dry-run --json --force
{"source":"api","replicaFallbackReason":"replica-absent"}
```

## One defect worth naming

The module exposes `LINEAR_IDENTITY_VALUE` because `$(fn)` is a **subshell** — the first
cut captured that way, so `LINEAR_IDENTITY_REASON` died with the subshell and the note
printed `(unknown)` for an absent replica. A failure that reports, but reports nothing
usable. This repo hit the identical defect in `setup-catalyst.sh` the same night (#3500);
both are pinned by tests now.

## Cross-validation

With the replica path and the forced-API path both run against the live workspace, the two
independently return the **same** `teamId` and the same 12 state UUIDs — so the replica's
answer is checked against Linear's own API, not just against itself.

## Tests

- `__tests__/linear-identity.test.sh` — **20** assertions over a fixture replica. Every
  positive is paired with its degraded case over the *same* call, and the empty-but-fresh
  case carries a positive control proving the freshness gate is open (so the failure is
  the empty table, not a broken gate). Includes SQL-quote escaping.
- `__tests__/resolve-linear-ids.test.sh` — **+4**, pinning the `--json` stream contract.
  **34 passed / 0 failed**, against a **measured `origin/main` baseline of 30/0** on the
  same harness (the 3 failures the first cut introduced were attributed by running that
  baseline, not assumed pre-existing).
- `linear-transition.test.sh`, `check-project-setup.test.sh` (the other
  `linear-state-ids.json` consumers): pass unchanged.
- `trunk check`: my files clean. `resolve-linear-ids.sh` and its test remain flagged as
  unformatted — **both are unformatted on `origin/main`** (verified against a pristine
  checkout), and reformatting them would bury this change under a whole-file sweep.

## Not in this increment

The 14 non-test callers of `getReplicaDbPath()`/`linear-state-ids.json` across JS **and**
bash. That cutover needs the one-registry / two-engines / parity-suite discipline of
`lib/secret-contract.mjs`, or a bash reader silently reads another tenant's replica — which
is worse than failing, because it returns plausible data.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugins/dev/scripts/lib/linear-identity.sh [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+# linear_identity_team_id <TEAM_KEY> [db] — echo the team UUID.
+#
+# ⚠️ Derived from the issues the replica carries for that key. A team with ZERO issues
+# is therefore not resolvable here, and that is reported as `team-not-resolvable`
+# rather than as "no such team": the two are indistinguishable from this source, and
+# claiming the stronger one would send an operator to re-create a team that exists.
+linear_identity_team_id() {
+	local key="$1" db="${2:-$CATALYST_REPLICA_DB}" id
+	_lid_ready "$db" || return 1
+	id=$(_lid_q "$db" "SELECT team_id FROM issues WHERE team_key='${key//\'/\'\'}' AND team_id IS NOT NULL AND removed_at IS NULL LIMIT 1;")
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply plugins/dev/scripts/lib/linear-identity.sh]: Correct, and it is the failure mode this module exists to prevent — not a missed error but a **confident wrong answer**: a nonempty-but-incomplete state map, or a duplicated label transiently looking unique. Thank you for the pointer to `linear-write-proxy-resolve.mjs`; it does resolve this the same way and I should have matched it.

Fixed in `f3e7e7437`. The cursor gate and the read now run as one statement (`SELECT CASE WHEN NOT EXISTS(cursor) THEN <sentinel> ELSE COALESCE((<read>), '') END`), so they share one snapshot. `_lid_ready`'s cheap early-out is kept but the comment now says explicitly that it is no longer the authoritative gate — a gate that passes on one connection says nothing about what a later one observes.

The race itself cannot be reproduced deterministically from a shell test, so the in-snapshot branch is tested **directly** instead: `replica_fresh` is stubbed to pass and the read must still refuse. Without that, `_LID_NO_CURSOR` would be a protection nobody has ever seen fire.

Related: `replica_fresh` folds a stale heartbeat and an absent cursor into one rc, and those need different repairs — they are now named apart (`replica-stale` vs `replica-reseeding`).

File: plugins/dev/scripts/lib/linear-identity.sh [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+	LINEAR_IDENTITY_VALUE="$id"
+	printf '%s' "$id"
+}
+
+# linear_identity_state_id <TEAM_KEY> <STATE_NAME> [db] — echo the workflow-state UUID.
+# Archived states are excluded: resolving to one produces a write Linear accepts and a
+# board nobody sees.
+linear_identity_state_id() {
+	local key="$1" name="$2" db="${3:-$CATALYST_REPLICA_DB}" team id
+	team=$(linear_identity_team_id "$key" "$db") || return 1
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply plugins/dev/scripts/lib/linear-identity.sh]: Fixed rather than deferred — this is the module's core contract, and you are right that it was defeated at the two helpers most callers actually use. `f3e7e7437`: both now call `linear_identity_team_id` directly and read `LINEAR_IDENTITY_VALUE`. Four new assertions cover nested absent / stale / unresolved-team reasons through `state_id` and `states_json`.

⛔ Worth recording, because my fix for this shipped the same defect one level down: `_lid_query_gated` was itself captured with `$( )`, so its own `_lid_fail` calls set a reason that died in that subshell — which is exactly why the new missing-table test failed on its first run with an empty reason. That is the **fourth** occurrence of this defect tonight and the **third inside a fix for it** (the others being `catalyst_helper_path` and `resolve-linear-ids.sh`'s replica branch, both in #3500). The rule, written down so it stops recurring: in bash a function cannot return both a value and a diagnosis through stdout — the value goes in a global and the caller invokes directly.

File: plugins/dev/scripts/lib/linear-identity.sh [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+		return 1
+	}
+	replica_fresh "$db" || {
+		_lid_fail "replica-stale"
+		return 1
+	}
+	LINEAR_IDENTITY_REASON=""
+}
+
+_lid_q() { sqlite3 "$1" "$2" 2>/dev/null; }
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply plugins/dev/scripts/lib/linear-identity.sh]: Fixed rather than deferred — "could not query" being reported as "the entity is not there" is precisely the false-clean this module is written against, so leaving it would have undercut the whole thing. And the live condition you cite is the right one: `workflow_states` absent on a replica whose gate is open.

`f3e7e7437`: the query status is preserved and surfaces as its own named failure, `replica-unreadable`, distinct from `team-not-resolvable` / `state-not-found` / `label-not-found`.

Tests assert a dropped `workflow_states` yields `replica-unreadable` from both `state_id` and `states_json`, with a **control on the same fixture** — the team read, whose table still exists, still resolves — so the failure is the dropped table and not a globally broken fixture.
